codetransformer/code.py

```diff
@@ -0,0 +1,432 @@
+from dis import Bytecode, HAVE_ARGUMENT
+from enum import IntEnum, unique
+from functools import reduce
+import operator
+from types import CodeType
+
+from .instructions import Instruction, LOAD_CONST
+from .utils.functional import scanl
+
+
+@unique
+class Flags(IntEnum):
+    CO_OPTIMIZED = 0x0001
+    CO_NEWLOCALS = 0x0002
+    CO_VARARGS = 0x0004
+    CO_VARKEYWORDS = 0x0008
+    CO_NESTED = 0x0010
+    CO_GENERATOR = 0x0020
+
+    # The CO_NOFREE flag is set if there are no free or cell variables.
+    # This information is redundant, but it allows a single flag test
+    # to determine whether there is any extra work to be done when the
+    # call frame it setup.
+    CO_NOFREE = 0x0040
+
+    # The CO_COROUTINE flag is set for coroutine functions (defined with
+    # ``async def`` keywords)
+    CO_COROUTINE = 0x0080
+    CO_ITERABLE_COROUTINE = 0x0100
+
+
+def _sparse_args(instrs):
+    """
+    Makes the arguments sparse so that instructions live at the correct
```

Comment:
I will update the docstring, but the idea is that jumps use the bytecode offset as the argument, sparse arg adds 2 Nones into the instructions list to keep the indexes the same